---
output:
  knitrBootstrap::bootstrap_document:
    title: "Compare transcript divergence with translation divergence"
    theme: default
    theme.chooser: TRUE
    highlight.chooser: TRUE
---


Differential divergence analysis
=================================

```{r run_author, results='asis', echo=FALSE}
email <- "<a href='mailto:joyce.hsiao1@gmail.com'>Joyce Hsiao</a>"
cat(paste("Author:", email))
```


```{r run_date, results='asis', echo=FALSE}
#last_update = format(Sys.time(), "(<time>%Y-%m-%d</time>)")
last_update <- Sys.time()
cat(paste("Last updated:", last_update))
```


[view source](compare-ribo-rna-divergence.Rmd)


```{r knitr_settings, include=FALSE, echo=FALSE}
library(knitr)
opts_knit$set(progress = TRUE, verbose = TRUE)
opts_chunk$set(message = FALSE, warning = FALSE, eval = TRUE, 
               echo = TRUE,
              bootstrap.panel = TRUE, 
              bootstrap.show.code = TRUE)
```



```{r load_packages}
library(Biobase)

## Set working directories 
dir <- "../"
figdir <- file.path(dir,"figures")
rdadir <- file.path(dir,"rdas")
datadir <- file.path(dir, "data")

## Use broman package for colors
require(broman)
crayon <- brocolors("crayons")

## Load customized packages
require(devtools)
require(Humanzee)
```




Human versus Chimp
==================


> We used a likelihood ratio test to compare species divergence 
at the Ribo level versus at the RNA level.


```{r, eval = F}
load(file.path(rdadir,"eSetRRP.rda"))

# eSetRRP.RP.Q.log2 includes log2 quantile-normalized RPKM values
# Take a human and rhesus data that includes only RNA and Ribo
eSet <- eSetRRP.RP.Q.log2[ , 
          eSetRRP.RP.Q.log2$seqData != "protein"]

# Interaction analysis comparing the inter-species
# fold change between RNA and Ribo
res.riborna <- interact2way(eSet)

save(res.riborna,file=file.path(rdadir,"TEnew.rda"))
```


> The genes with significant interaction effect at FDR < .01 are identified
as significant in species difference between RNA and Ribo, or equivalently,
species difference in translation efficiency. 


```{r}
load(file.path(rdadir, "TEnew.rda"))

# Number of genes with sig. species difference in TE
sum(res.riborna$int.qval < .01)
```


Other inter-species comparisons
=================================

> We performed the same likelihood ratio test to compare RNA and Ribo layer
divergence between rhesus monkey and human or chimpanzee.

```{r, eval = F}
# load eSetRRP.RP.Q.log2, which contains log2 RPKM values, quantile-normalized
load(file.path(rdadir,"eSetRRP.rda"))

# Subset human and rhesus samples in RNA and Ribo data
eSet_human_rhesus <- 
    eSetRRP.RP.Q.log2[, eSetRRP.RP.Q.log2$species != "chimp" & 
                        eSetRRP.RP.Q.log2$seqData != "protein"]

# Interaction analysis comparing RNA-Ribo between human and rhesus
res.riborna_human_rhesus <- interact2way(eSet_human_rhesus)

# Subset chimp and rhesus samples in RNA and Ribo data
eSet_chimp_rhesus <- 
    eSetRRP.RP.Q.log2[, eSetRRP.RP.Q.log2$species != "human" & 
                        eSetRRP.RP.Q.log2$seqData != "protein"]

# Interaction analysis comparing RNA-Ribo between chimp and rhesus
res.riborna_chimp_rhesus <- interact2way_full_buffer(eSet_chimp_rhesus) 


save(res.riborna_human_rhesus, res.riborna_chimp_rhesus,
     file = file.path(rdadir,"TE_other_interspecies.rda"))
```

> Number of genes with sig. TE difference between rhesus and human or rhesus.


```{r}
load(file.path(rdadir, "TE_other_interspecies.rda"))

data.frame(human_rhesus = sum(res.riborna_human_rhesus$int.qval < .01),
           chimp_rhesus = sum(res.riborna_chimp_rhesus$int.qval < .01) )
```





QC checks on the results above
=================================

> The above results used a subset of RNA and Ribo data that was 
already quantile-normalized. Therefore, the distribution of these
3,188 genes are not exactly the same across the samples. 

> We quantile-normalized sequencing data of 3,188 genes for each of the
data types across the three species. 


```{r, eval = FALSE}
load(file.path(rdadir,"eSetRRP.rda"))

# Quantile normalize RNA RPKM values across species
qmatRNA <- normalize(exprs(eSetRRP.RP[,eSetRRP.RP$seqData=="rna"]),
                     geneLen = NULL,
                     method="quantile",log="TRUE")

# Quantile normalize Ribo RPKM values across species
qmatRIBO <- normalize(exprs(eSetRRP.RP[,eSetRRP.RP$seqData=="ribo"]),
                      geneLen = NULL,
                      method="quantile",log="TRUE")

# Quantile normalize mass spectrametry data across species
qmatMASS <- normalize(exprs(eSetRRP.RP[,eSetRRP.RP$seqData=="protein"]),
                      geneLen = NULL,
                      method="quantile",log="FALSE")

# Make expression set 
emat <- cbind(qmatRNA, qmatRIBO, qmatMASS)
eSetRRP.RP.Q.log2.post <- ExpressionSet(assayData = emat,
                       phenoData=phenoData(eSetRRP.RP),
                       experimentData=experimentData(eSetRRP.RP))
featureData(eSetRRP.RP.Q.log2.post) <- featureData(eSetRRP.RP.Q.log2.post)     
```

## Inter-species comparisons


```{r, eval = F}
# Subset human and chimp samples in RNA and Ribo data
eSet.post_human_chimp <- 
  eSetRRP.RP.Q.log2.post[ , eSetRRP.RP.Q.log2.post$species != "rhesus" & 
                     eSetRRP.RP.Q.log2.post$seqData != "protein"]

# Subset human and rhesus samples in RNA and Ribo data
eSet.post_human_rhesus <- 
  eSetRRP.RP.Q.log2.post[ , eSetRRP.RP.Q.log2.post$species != "chimp" & 
                     eSetRRP.RP.Q.log2.post$seqData != "protein"]

# Subset chimp and rhesus samples in RNA and Ribo data
eSet.post_chimp_rhesus <- 
  eSetRRP.RP.Q.log2.post[ , eSetRRP.RP.Q.log2.post$species != "human" & 
                     eSetRRP.RP.Q.log2.post$seqData != "protein"]

# Likelihood ratio test for human vs. chimp difference in 
# translation efficiency
res.riborna.post_human_chimp <- interact2way(eSet.post_human_chimp)

# Likelihood ratio test for human vs. rhesus difference in 
# translation efficiency
res.riborna.post_human_rhesus <- interact2way(eSet.post_human_rhesus)

# Likelihood ratio test for chimp vs. rhesus difference in 
# translation efficiency
res.riborna.post_chimp_rhesus <- interact2way(eSet.post_chimp_rhesus)


save(eSetRRP.RP.Q.log2.post,
     res.riborna.post_human_chimp,
     res.riborna.post_human_rhesus, 
     res.riborna.post_chimp_rhesus,
     file = file.path(rdadir,"TE_postQ.rda"))
```

> Number of genes sig. difference in translation efficiency 

```{r}
load(file.path(rdadir, "TE_postQ.rda"))
data.frame(human_chimp = sum(res.riborna.post_human_chimp$int.qval < .01),
           human_rhesus = sum(res.riborna.post_human_rhesus$int.qval < .01),
           chimp_rhesus = sum(res.riborna.post_chimp_rhesus$int.qval < .01) )
```



SessionInfo
===========

```{r}
sessionInfo()
```

