<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <title></title>
  <style type="text/css">code{white-space: pre;}</style>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <!-- jQuery -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/jqueryui/1.10.3/jquery-ui.min.js"></script>
  
  <!-- bootstrap -->
  <!--<link href="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css" rel="stylesheet"  id="style">-->
  <script src="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/js/bootstrap.min.js"></script>
  
  <!-- highlight.js -->
  <!--<link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/default.min.css" rel="stylesheet" id="code-style">-->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/highlight.min.js"></script>
  <script>
  hljs.LANGUAGES.r=function(a){var b="([a-zA-Z]|\\.[a-zA-Z.])[a-zA-Z0-9._]*";return{c:[a.HCM,{b:b,l:b,k:{keyword:"function if in break next repeat else for return switch while try tryCatch|10 stop warning require library attach detach source setMethod setGeneric setGroupGeneric setClass ...|10",literal:"NULL NA TRUE FALSE T F Inf NaN NA_integer_|10 NA_real_|10 NA_character_|10 NA_complex_|10"},r:0},{cN:"number",b:"0[xX][0-9a-fA-F]+[Li]?\\b",r:0},{cN:"number",b:"\\d+(?:[eE][+\\-]?\\d*)?L\\b",r:0},{cN:"number",b:"\\d+\\.(?!\\d)(?:i\\b)?",r:0},{cN:"number",b:"\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",r:0},{b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[a.BE],r:0},{cN:"string",b:"'",e:"'",c:[a.BE],r:0}]}}(hljs); </script>
  <!--<script type="text/javascript", src="https://yandex.st/highlightjs/7.3/languages/r.min.js"></script>-->
  
  <!-- Manific Popup -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/magnific-popup.js/0.8.9/jquery.magnific-popup.min.js"></script>
  
  <script type="text/javascript"
    src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  </script>
  <script defer="defer">
  // Function to generate the dynamic table of contents
  jQuery.fn.generate_TOC = function () {
    var base = $(this[0]);
  
    var selectors = ['h1', 'h2', 'h3', 'h4'];
  
    var last_ptr = [{}, {}, {}, {}];
  
    var anchors = {};
  
    generate_anchor = function (text) {
      var test = text.replace(/\W/g, '_');
  
      while(test in anchors){
        //if no suffix, add one
        if(test.match(/_\d+$/) === null){
          test = test + "_2";
        }
        //else generate unique id for duplicates by adding one to the suffix
        else {
          test = test.replace(/_(\d+)$/, function(match, number){ var num=+number+1; return("_" + num) });
        }
      }
      anchors[test]=1;
      return(test);
    }
  
    $(selectors.join(',')).filter(function(index) { return $(this).parent().attr("id") != 'header'; }).each(function () {
  
      var heading = $(this);
      var idx = selectors.indexOf(heading.prop('tagName').toLowerCase());
      var itr = 0;
  
      while (itr <= idx) {
        if (jQuery.isEmptyObject(last_ptr[itr])) {
          last_ptr[itr] = $('<ul>').addClass('nav');
          if (itr === 0) {
            base.append(last_ptr[itr])
          } else {
            if(last_ptr[itr-1].children('li').length === 0){
              last_ptr[itr-1].append(last_ptr[itr]);
            }
            else {
              last_ptr[itr - 1].children('li').last().append(last_ptr[itr]);
            }
          }
        }
        itr++;
      }
      var anchor = generate_anchor(heading.text());
      heading.attr('id', anchor);
      var a = $('<a>')
      .text(heading.text())
      .attr('href', '#' + anchor);
  
    var li = $('<li>')
      .append(a);
  
    last_ptr[idx].append(li);
    for (i = idx + 1; i < last_ptr.length; i++) {
      last_ptr[i] = {};
    }
    });
  }
  /* run scripts when document is ready */
  $(function() {
    "use strict";
  
    var $window = $(window);
    var $body = $(document.body);
  
    /* size of thumbnails */
  
    var hidden_types = ['source']
    var output_types = ['output', 'message', 'warning', 'error']
  
    /* style tables */
    $('table').addClass('table table-striped table-bordered table-hover table-condensed');
  
    $('pre code').each(function(i, e) {
      hljs.highlightBlock(e);
    });
  
    /* Magnific Popup */
    $(".thumbnail").each(function(){
      $(this).magnificPopup({
        disableOn: 768,
        closeOnContentClick: true,
  
        type: 'image',
        items: {
          src: $(this).find('img').attr('src'),
        }
      });
    });
  
    function toggle_block(obj, show) {
      var span = obj.find('span');
      if(show === true){
        span.removeClass('glyphicon-chevron-up').addClass('glyphicon-chevron-down');
        obj.next('pre').slideDown();
      }
      else {
        span.removeClass('glyphicon-chevron-down').addClass('glyphicon-chevron-up');
        obj.next('pre').slideUp();
      }
    }
  
    function toggle_thumbnails(imgs, show){
      if(show === true){
        imgs.parents().show()
        imgs.slideDown();
      }
      else {
        imgs.slideUp(400, function(){ $(this).parent().hide(); });
      }
    }
  
    function global_toggle(obj){
      var type = obj.attr('type');
      var show = !obj.parent('li').hasClass('active');
      if(show === true){
        obj.parent('li').addClass('active');
      }
      else{
        obj.parent('li').removeClass('active');
      }
      if(type == 'figure'){
        toggle_thumbnails($('.thumbnail img'), show);
      }
      else {
        $('.toggle.' + type).each(function() { toggle_block($(this), show); });
      }
    }
  
    /* onclick toggle next code block */
    $('.toggle').click(function() {
      var span = $(this).find('span');
      toggle_block($(this), !span.hasClass('glyphicon-chevron-down'));
      return false
    })
  
    // global toggles
    $('.toggle-global').click(function(){
      var type = $(this).attr('type');
      if(type === 'all-source'){
          $('li a.source').each(function() {
            global_toggle($(this));
          });
        }
      else if(type === 'all-output'){
        $.each(output_types, function(i, val){
          console.log(val);
          global_toggle($('li a.' + val));
        });
      }
      else {
        console.log($(this));
        global_toggle($(this));
      }
      return false;
    });
    /* table of contents */
    if($(['h1', 'h2', 'h3', 'h4'].join(',')).length > 0){
      $('body > #wrap > .container > .row').append('<div class="col-md-2"><div id="toc" class="well sidebar sidenav affix hidden-print"/></div>');
      $('#toc').generate_TOC();
    }
  
    $.each(hidden_types, function(i, type) {
      $('li[type=' + type + ']').each(function(){ global_toggle($(this)); });
    });
  
    /* remove paragraphs with no content */
    $('p:empty').remove();
  
    $body.scrollspy({
      target: '.sidebar',
    });
  
    /* theme switch */
    $('.theme-switch').click(function(){
      var css = $('link[title=' + $(this).attr('title') + ']');
      $('#theme[rel=stylesheet]').attr('href', css.attr('href'));
      $('.theme-switch').closest('li').removeClass('active');
      $(this).closest('li').addClass('active');
      return false;
    });
    /* code style switch */ //TODO use same function for both of these?
    $('.highlight-switch').click(function(){
      var css = $('link[title="' + $(this).attr('title') + '"]');
      $('#highlight[rel=stylesheet]').attr('href', css.attr('href'));
      $('.highlight-switch').closest('li').removeClass('active');
      $(this).closest('li').addClass('active');
      return false;
    });
  
    //TODO refresh on show/hide
    $window.on('load', function () {
      $body.scrollspy('refresh');
    })
  
  });
  
  </script>
  <style>
  /* Knitr_bootstrap styles */
  #header {
    display: none !important;
    visibility: hidden !important;
  }
  #wrap .container-fluid {
    padding: 0;
    overflow: hidden;
  }
  .toggle{
    text-transform: capitalize;
  }
  
  .toggle-global{
    text-transform: capitalize;
  }
  
  /* Sticky footer styles */
  * {
    margin:0;
  }
  html,
  body {
      height: 100%;
      padding:0 !important;
      /* The html and body elements cannot have any padding or margin. */
      /*overflow-x: hidden;*/
  }
  
  /* Wrapper for page content to push down footer */
  #wrap {
      min-height: 100%;
      height: auto !important;
      height: 100%;
      /* Negative indent footer by it's height */
      margin: 0 auto -120px;
  }
  
  /* Set the fixed height of the footer here */
  #push,
  #footer {
      height: 120px;
  }
  
  #footer {
    text-align: center;
  }
  
  /* Top level subheader elements.  These are the first nested items underneath a header element. */
  .header li {
    font-size: 20px;
  }
  
  /* Makes the font smaller for all subheader elements. */
  .sub-header li {
      font-size: 12px;
  }
  
  button.thumbnails {
    margin-left:0px;
  }
  
  /*
   * Side navigation
   *
   * Scrollspy and affixed enhanced navigation to highlight sections and secondary
   * sections of docs content.
   */
  
  /* By default it's not affixed in mobile views, so undo that */
  .sidebar.affix {
    position: static;
  }
  
  /* First level of nav */
  .sidenav {
    margin-top: 30px;
    margin-bottom: 30px;
    padding-top:    10px;
    padding-bottom: 10px;
    border-radius: 5px;
  }
  
  /* All levels of nav */
  .sidebar .nav > li > a {
    display: block;
    padding: 5px 20px;
  }
  .sidebar .nav > li > a:hover,
  .sidebar .nav > li > a:focus {
    text-decoration: none;
    border-right: 1px solid;
  }
  .sidebar .nav > .active > a,
  .sidebar .nav > .active:hover > a,
  .sidebar .nav > .active:focus > a {
    font-weight: bold;
    background-color: transparent;
    border-right: 1px solid;
  }
  
  /* Nav: second level (shown on .active) */
  .sidebar .nav .nav {
    display: none; /* Hide by default, but at >768px, show it */
    margin-bottom: 8px;
  }
  .sidebar .nav .nav > li > a {
    padding-top:    3px;
    padding-bottom: 3px;
    padding-left: 30px;
    font-size: 90%;
  }
  
  .sidebar .nav .nav .nav > li > a {
    padding-left: 40px;
  }
  .sidebar .nav .nav .nav .nav > li > a {
    padding-left: 50px;
  }
  
  /* Show and affix the side nav when space allows it */
  @media screen and (min-width: 992px) {
    .sidebar .nav > .active > ul {
      display: block;
    }
    /* Widen the fixed sidebar */
    .sidebar.affix,
    .sidebar.affix-bottom {
      width: 213px;
    }
    .sidebar.affix-top,
    .sidebar.affix {
      position: fixed; /* Undo the static from mobile first approach */
      top: 30px;
    }
    .sidebar.affix-bottom {
      position: absolute; /* Undo the static from mobile first approach */
    }
    .sidebar.affix-bottom .sidenav,
    .sidebar.affix .sidenav {
      margin-top: 0;
      margin-bottom: 0;
    }
  }
  @media screen and (min-width: 1200px) {
    /* Widen the fixed sidebar again */
    .sidebar.affix-bottom,
    .sidebar.affix {
      width: 263px;
    }
  }
  
  #toc {
    padding: 10px 0px;
    margin:0;
    border:0;
  }
  
  
  .panel pre {
    margin: 0;
    padding: 0;
    border: 0;
  }
  button + pre {
    margin: 0;
    padding: 0;
  }
  pre code {
    border-radius: 0;
  }
  /* Magnific Popup CSS */
  .mfp-bg {
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 1042;
    overflow: hidden;
    position: fixed;
    background: #0b0b0b;
    opacity: 0.8;
    filter: alpha(opacity=80); }
  
  .mfp-wrap {
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 1043;
    position: fixed;
    outline: none !important;
    -webkit-backface-visibility: hidden; }
  
  .mfp-container {
    text-align: center;
    position: absolute;
    width: 100%;
    height: 100%;
    left: 0;
    top: 0;
    padding: 0 8px;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box; }
  
  .mfp-container:before {
    content: '';
    display: inline-block;
    height: 100%;
    vertical-align: middle; }
  
  .mfp-align-top .mfp-container:before {
    display: none; }
  
  .mfp-content {
    position: relative;
    display: inline-block;
    vertical-align: middle;
    margin: 0 auto;
    text-align: left;
    z-index: 1045; }
  
  .mfp-inline-holder .mfp-content,
  .mfp-ajax-holder .mfp-content {
    width: 100%;
    cursor: auto; }
  
  .mfp-ajax-cur {
    cursor: progress; }
  
  .mfp-zoom-out-cur,
  .mfp-zoom-out-cur .mfp-image-holder .mfp-close {
    cursor: -moz-zoom-out;
    cursor: -webkit-zoom-out;
    cursor: zoom-out; }
  
  .mfp-zoom {
    cursor: pointer;
    cursor: -webkit-zoom-in;
    cursor: -moz-zoom-in;
    cursor: zoom-in; }
  
  .mfp-auto-cursor .mfp-content {
    cursor: auto; }
  
  .mfp-close,
  .mfp-arrow,
  .mfp-preloader,
  .mfp-counter {
    -webkit-user-select: none;
    -moz-user-select: none;
    user-select: none; }
  
  .mfp-loading.mfp-figure {
    display: none; }
  
  .mfp-hide {
    display: none !important; }
  
  .mfp-preloader {
    color: #cccccc;
    position: absolute;
    top: 50%;
    width: auto;
    text-align: center;
    margin-top: -0.8em;
    left: 8px;
    right: 8px;
    z-index: 1044; }
  
  .mfp-preloader a {
    color: #cccccc; }
  
  .mfp-preloader a:hover {
    color: white; }
  
  .mfp-s-ready .mfp-preloader {
    display: none; }
  
  .mfp-s-error .mfp-content {
    display: none; }
  
  button.mfp-close,
  button.mfp-arrow {
    overflow: visible;
    cursor: pointer;
    background: transparent;
    border: 0;
    -webkit-appearance: none;
    display: block;
    padding: 0;
    z-index: 1046;
    -webkit-box-shadow: none;
    box-shadow: none; }
  
  button::-moz-focus-inner {
    padding: 0;
    border: 0; }
  
  .mfp-close {
    width: 44px;
    height: 44px;
    line-height: 44px;
    position: absolute;
    right: 0;
    top: 0;
    text-decoration: none;
    text-align: center;
    opacity: 0.65;
    padding: 0 0 18px 10px;
    color: white;
    font-style: normal;
    font-size: 28px;
    font-family: Arial, Baskerville, monospace; }
    .mfp-close:hover, .mfp-close:focus {
      opacity: 1; }
    .mfp-close:active {
      top: 1px; }
  
  .mfp-close-btn-in .mfp-close {
    color: #333333; }
  
  .mfp-image-holder .mfp-close,
  .mfp-iframe-holder .mfp-close {
    color: white;
    right: -6px;
    text-align: right;
    padding-right: 6px;
    width: 100%; }
  
  .mfp-counter {
    position: absolute;
    top: 0;
    right: 0;
    color: #cccccc;
    font-size: 12px;
    line-height: 18px; }
  
  .mfp-arrow {
    position: absolute;
    opacity: 0.65;
    margin: 0;
    top: 50%;
    margin-top: -55px;
    padding: 0;
    width: 90px;
    height: 110px;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0); }
  
  .mfp-arrow:active {
    margin-top: -54px; }
  
  .mfp-arrow:hover,
  .mfp-arrow:focus {
    opacity: 1; }
  
  .mfp-arrow:before, .mfp-arrow:after,
  .mfp-arrow .mfp-b,
  .mfp-arrow .mfp-a {
    content: '';
    display: block;
    width: 0;
    height: 0;
    position: absolute;
    left: 0;
    top: 0;
    margin-top: 35px;
    margin-left: 35px;
    border: medium inset transparent; }
  .mfp-arrow:after,
  .mfp-arrow .mfp-a {
    border-top-width: 13px;
    border-bottom-width: 13px;
    top: 8px; }
  .mfp-arrow:before,
  .mfp-arrow .mfp-b {
    border-top-width: 21px;
    border-bottom-width: 21px; }
  
  .mfp-arrow-left {
    left: 0; }
    .mfp-arrow-left:after,
    .mfp-arrow-left .mfp-a {
      border-right: 17px solid white;
      margin-left: 31px; }
    .mfp-arrow-left:before,
    .mfp-arrow-left .mfp-b {
      margin-left: 25px;
      border-right: 27px solid #3f3f3f; }
  
  .mfp-arrow-right {
    right: 0; }
    .mfp-arrow-right:after,
    .mfp-arrow-right .mfp-a {
      border-left: 17px solid white;
      margin-left: 39px; }
    .mfp-arrow-right:before,
    .mfp-arrow-right .mfp-b {
      border-left: 27px solid #3f3f3f; }
  
  .mfp-iframe-holder {
    padding-top: 40px;
    padding-bottom: 40px; }
  
  .mfp-iframe-holder .mfp-content {
    line-height: 0;
    width: 100%;
    max-width: 900px; }
  
  .mfp-iframe-scaler {
    width: 100%;
    height: 0;
    overflow: hidden;
    padding-top: 56.25%; }
  
  .mfp-iframe-scaler iframe {
    position: absolute;
    display: block;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    box-shadow: 0 0 8px rgba(0, 0, 0, 0.6);
    background: black; }
  
  .mfp-iframe-holder .mfp-close {
    top: -40px; }
  
  /* Main image in popup */
  img.mfp-img {
    width: auto;
    max-width: 100%;
    height: auto;
    display: block;
    line-height: 0;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
    padding: 40px 0 40px;
    margin: 0 auto; }
  
  /* The shadow behind the image */
  .mfp-figure:after {
    content: '';
    position: absolute;
    left: 0;
    top: 40px;
    bottom: 40px;
    display: block;
    right: 0;
    width: auto;
    height: auto;
    z-index: -1;
    box-shadow: 0 0 8px rgba(0, 0, 0, 0.6);
    background: #444444; }
  
  .mfp-figure {
    line-height: 0; }
  
  .mfp-bottom-bar {
    margin-top: -36px;
    position: absolute;
    top: 100%;
    left: 0;
    width: 100%;
    cursor: auto; }
  
  .mfp-title {
    text-align: left;
    line-height: 18px;
    color: #f3f3f3;
    word-wrap: break-word;
    padding-right: 36px; }
  
  .mfp-figure small {
    color: #bdbdbd;
    display: block;
    font-size: 12px;
    line-height: 14px; }
  
  .mfp-image-holder .mfp-content {
    max-width: 100%; }
  
  .mfp-gallery .mfp-image-holder .mfp-figure {
    cursor: pointer; }
  
  @media screen and (max-width: 800px) and (orientation: landscape), screen and (max-height: 300px) {
    /**
     * Remove all paddings around the image on small screen
     */
    .mfp-img-mobile .mfp-image-holder {
      padding-left: 0;
      padding-right: 0; }
  
    .mfp-img-mobile img.mfp-img {
      padding: 0; }
  
    /* The shadow behind the image */
    .mfp-img-mobile .mfp-figure:after {
      top: 0;
      bottom: 0; }
  
    .mfp-img-mobile .mfp-bottom-bar {
      background: rgba(0, 0, 0, 0.6);
      bottom: 0;
      margin: 0;
      top: auto;
      padding: 3px 5px;
      position: fixed;
      -webkit-box-sizing: border-box;
      -moz-box-sizing: border-box;
      box-sizing: border-box; }
  
    .mfp-img-mobile .mfp-bottom-bar:empty {
      padding: 0; }
  
    .mfp-img-mobile .mfp-counter {
      right: 5px;
      top: 3px; }
  
    .mfp-img-mobile .mfp-close {
      top: 0;
      right: 0;
      width: 35px;
      height: 35px;
      line-height: 35px;
      background: rgba(0, 0, 0, 0.6);
      position: fixed;
      text-align: center;
      padding: 0; }
  
    .mfp-img-mobile .mfp-figure small {
      display: inline;
      margin-left: 5px; } }
  @media all and (max-width: 900px) {
    .mfp-arrow {
      -webkit-transform: scale(0.75);
      transform: scale(0.75); }
  
    .mfp-arrow-left {
      -webkit-transform-origin: 0;
      transform-origin: 0; }
  
    .mfp-arrow-right {
      -webkit-transform-origin: 100%;
      transform-origin: 100%; }
  
    .mfp-container {
      padding-left: 6px;
      padding-right: 6px; } }
  .mfp-ie7 .mfp-img {
    padding: 0; }
  .mfp-ie7 .mfp-bottom-bar {
    width: 600px;
    left: 50%;
    margin-left: -300px;
    margin-top: 5px;
    padding-bottom: 5px; }
  .mfp-ie7 .mfp-container {
    padding: 0; }
  .mfp-ie7 .mfp-content {
    padding-top: 44px; }
  .mfp-ie7 .mfp-close {
    top: 0;
    right: 0;
    padding-top: 0; }
  
  //Magnific overrides
  .mfp-image img{
    background: white;
  }
  .mfp-figure:after {
    background: white;
  }
  
  /*
   * Off Canvas navbar toggle right
   * --------------------------------------------------
   */
  
  @media screen and (max-width: 768px) {
    .row-offcanvas .collapsing {
    -webkit-transition: none 0;
      -moz-transition: none 0;
      transition: none 0;
    }
   .row-offcanvas .navbar {
    position: absolute;
    z-index: 2;
      right:0;
      height:100%;
      width:55px;
      border:0;
      background-color:transparent;
    }
    .row-offcanvas .navbar-toggle {
      margin-right: 5px;
      margin-left: 5px;
    }
    .row-offcanvas {
      position: relative;
    }
    .row-offcanvas-right.active .navbar {
    position: absolute;
    z-index: 2;
      right: -28.4%;
      width:40%;
      background-color:#eee;
      border:0 solid #ddd;
      border-left-width:1px;
    }
    .row-offcanvas-right.active {
      left: -30%;
    }
    .row-offcanvas-right.active .navbar-collapse {
      position: relative;
      width: 100%;
    }
    .row-offcanvas .content {
    /*width:calc(100% - 60px);*/
    }
  }
  </style>
</head>
<body>
<title>
RNA-Protein divergence conditioned on RNA-Ribo divergence
</title>
<div id="wrap">
<div class="container">
<div class="row row-offcanvas row-offcanvas-right">
<div class="contents col-xs-12 col-md-10">
<h1 id="rna-protein-divergence-conditioned-on-rna-ribo-divergence">RNA-Protein divergence conditioned on RNA-Ribo divergence</h1>
<div class="row">
Author: <a href='mailto:joyce.hsiao1@gmail.com'>Joyce Hsiao</a>
</div>
<div class="row">
Last updated: 2015-08-21 02:08:20
</div>
<p><a href="translation-efficiency-protein-divergence-20150812.Rmd">view source</a></p>
<div class="row">
<div class="panel panel-primary">
<div class="source R panel-heading toggle">
<h5 class="panel-title">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</h5>
</div>
<pre style=""><code class="source r">library(Biobase)

## Set working directories 
dir <- "../"
figdir <- file.path(dir,"figures")
rdadir <- file.path(dir,"rdas")
datadir <- file.path(dir, "data")

## Use broman package for colors
require(broman)
crayon <- brocolors("crayons")

## Load customized packages
require(devtools)
require(Humanzee)</code></pre>
</div>
</div>
<h1 id="goal">Goal</h1>
<p>Consider the set of genes with sig. difference in translation efficiency between human and chimpanzee. We then categorize the genes into attenuation or enhancement at the translation level. Finally, we compute the proportion of genes attenuated at the protein level in the set of genes attenuated at the RNA level, and the proportion of genes enforced at protein level among the set of genes attenuated at the RNA level.</p>
<p>These analyses are for investigating whether divergence at the translation level is likely to be carried over to the protein level.</p>
<h1 id="definitions">Definitions</h1>
<p>Of the genes with sig. inter-species differences in translation efficiency, we considered the genes with significant inter-species difference at the RNA level and classified these genes as follows</p>
<ol style="list-style-type: decimal">
<li><p>Enhanced at Ribo</p>
<p>1.1 Of those enhanced at Ribo, No. of genes enhanced at protein</p></li>
<li><p>Attenuated at Ribo</p>
<p>2.1 Of those attenuated at Ribo, No. of genes attenuated at protein</p></li>
</ol>
<p>The definitions of enhancement and attenuation follows Definition version-20150812.</p>
<div class="row">
<div class="panel panel-primary">
<div class="source R panel-heading toggle">
<h5 class="panel-title">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</h5>
</div>
<pre style=""><code class="source r">## Load rdas of LRT results and compute fold change estimates using
## RPKM values before normalization. 

## load LRT results of RNA vs. Ribo divergence
load(file.path(rdadir,"TEnew.rda"))

## load LRT results of RNA vs. Protein divergence
load(file.path(rdadir,"rnapro.rda"))


## Compute fold changes based on un-normalized data
## from the object eSetRRP.log2 in eSetRRP.rda
load( file.path(rdadir,"eSetRRP.rda") )

eSet.temp <- eSetRRP.log2[ ,eSetRRP.log2$species!="rhesus"]

fc.mat <- lapply( seq_along( c("human","chimp") ), function(i) {
  ii <- eSet.temp$species == c("human","chimp")[i]
  eSet.tt <- eSet.temp[, ii]
  emat <- lapply( seq_along( c("ribo","rna","protein") ), function(j) {
    jj <- eSet.tt$seqData == c("ribo","rna","protein")[j]
    rowMeans( exprs(eSet.tt[, jj] ), na.rm=TRUE )
  })
  emat <- do.call(cbind, emat)
  colnames(emat) <- c("ribo","rna","protein")
  return( data.frame(emat) )
})
names(fc.mat) <- c("human","chimp")

dmat_unnormed <- data.frame( ribo = fc.mat$human$ribo-fc.mat$chimp$ribo,
                             rna = fc.mat$human$rna-fc.mat$chimp$rna,
                             pro = fc.mat$human$protein-fc.mat$chimp$protein )

xy.rnapro <- data.frame( rna = dmat_unnormed$rna,
                         pro = dmat_unnormed$pro )
xy.riborna <- data.frame( rna = dmat_unnormed$rna,
                          ribo = dmat_unnormed$ribo )
xy.ribopro <- data.frame( ribo = dmat_unnormed$ribo,
                          pro = dmat_unnormed$pro )</code></pre>
</div>
</div>
<h2 id="rna-vs.-ribo-divergence">RNA vs. Ribo divergence</h2>
<div class="row">
<div class="panel panel-primary">
<div class="source R panel-heading toggle">
<h5 class="panel-title">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</h5>
</div>
<pre style=""><code class="source r">## Genes with sig. human/chimp difference in divergence between RNA and Ribo
ii_sig_rnaribo <- res.riborna$int.qval < .01


## Category 1 - Enhanced divergence, criteria are as follows
## a) Sig. RNA vs. Ribo difference, AND
## b) ( abs(RNA FC) < abs(Ribo FC) ) AND ( RNA FC * Ribo FC > 0)
ii_riboGTrna <- (ii_sig_rnaribo) &  
                ( (abs(xy.riborna$ribo) > abs(xy.riborna$rna)) & 
                  (xy.riborna$rna * xy.riborna$ribo > 0 ) )


## Category 2 - Attenuated divergence, criteria are as follows
## a) Sig. RNA vs. Ribo diff., AND
## b) ( abs(RNA FC) > abs(Ribo FC) ) AND ( RNA FC * Ribo FC < 0)
ii_rnaGTribo <- (ii_sig_rnaribo) &  
                ( (abs(xy.riborna$rna) > abs(xy.riborna$ribo) ) | 
                  (xy.riborna$rna * xy.riborna$ribo < 0) ) 

Ribo_refmat <- rbind.data.frame(attenuated = sum(ii_rnaGTribo),
                                   enhanced = sum(ii_riboGTrna) )
colnames(Ribo_refmat) <- "# genes"
kable(Ribo_refmat)</code></pre>
</div>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right"># genes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">attenuated</td>
<td align="right">369</td>
</tr>
<tr class="even">
<td align="left">enhanced</td>
<td align="right">109</td>
</tr>
</tbody>
</table>
</div>
<h2 id="rna-vs.-protein-divergence">RNA vs. Protein divergence</h2>
<div class="row">
<div class="panel panel-primary">
<div class="source R panel-heading toggle">
<h5 class="panel-title">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</h5>
</div>
<pre style=""><code class="source r">## Genes with sig. human/chimp difference in divergence between RNA and protein
ii_sig_rnapro <- res.rnapro$int.qval < .01

## Category 1 - Enhanced divergence, criteria are as follows
## a) Sig. RNA vs. Protein difference, AND
## b) ( abs(RNA FC) < abs(Protein FC) ) AND ( RNA FC * Protein FC > 0)
ii_proGTrna <- (ii_sig_rnapro) &  
               ( (abs(xy.rnapro$pro) > abs(xy.rnapro$rna)) & 
                 (xy.rnapro$rna * xy.rnapro$pro > 0 ) )


## Category 2 - Attenuated divergence, criteria are as follows
## a) Sig. RNA vs. Protein diff., AND
## b) ( abs(RNA FC) > abs(Protein FC) ) OR ( RNA FC * Protein FC < 0)
ii_rnaGTpro <- (ii_sig_rnapro) &  
               ( (abs(xy.rnapro$rna) > abs(xy.rnapro$pro) ) | 
                 (xy.rnapro$rna * xy.rnapro$pro < 0) ) 


Protein_refmat <- rbind.data.frame( attenuated = sum(ii_rnaGTpro),
                                    enhanced = sum(ii_proGTrna) )
colnames(Protein_refmat) <- "# genes"

kable(Protein_refmat)</code></pre>
</div>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right"># genes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">attenuated</td>
<td align="right">294</td>
</tr>
<tr class="even">
<td align="left">enhanced</td>
<td align="right">17</td>
</tr>
</tbody>
</table>
</div>
<h1 id="gene-classfication">Gene classfication</h1>
<h2 id="enhancement-and-attenuation">Enhancement and attenuation</h2>
<div class="row">
<div class="panel panel-primary">
<div class="source R panel-heading toggle">
<h5 class="panel-title">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</h5>
</div>
<pre style=""><code class="source r">## Of the genes with sig. translation efficiency
## RNA-Ribo divergence versus RNA-protein divergence classification
summary_table_rnaribo_rnapro <- matrix(0, 2, 2,
                                    dimnames = list(rnaribo = c("rnaribo_ehanced",
                                                                "rnaribo_attenuated"),
                                                    rnapro = c("rnapro_enhanced",
                                                               "rnapro_attenuated") ) )
summary_table_rnaribo_rnapro[1, 1] <- sum(ii_riboGTrna & ii_proGTrna)
summary_table_rnaribo_rnapro[1, 2] <- sum(ii_riboGTrna & ii_rnaGTpro)
summary_table_rnaribo_rnapro[2, 1] <- sum(ii_rnaGTribo & ii_proGTrna)
summary_table_rnaribo_rnapro[2, 2] <- sum(ii_rnaGTribo & ii_rnaGTpro)

kable(summary_table_rnaribo_rnapro)</code></pre>
</div>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">rnapro_enhanced</th>
<th align="right">rnapro_attenuated</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">rnaribo_ehanced</td>
<td align="right">5</td>
<td align="right">14</td>
</tr>
<tr class="even">
<td align="left">rnaribo_attenuated</td>
<td align="right">0</td>
<td align="right">84</td>
</tr>
</tbody>
</table>
</div>
<h2 id="consistency-of-the-direction-of-divergence">Consistency of the direction of divergence</h2>
<div class="row">
<div class="panel panel-primary">
<div class="source R panel-heading toggle">
<h5 class="panel-title">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</h5>
</div>
<pre style=""><code class="source r">## Of the genes with sig. translation efficiency
## RNA-Ribo divergence versus RNA-protein divergence classification
summary_table_rnaribo_rnapro_consistency <- 
    matrix(0, 2, 2,
           dimnames = list(rnaribo = c("rnaribo_same_direction",
                                      "rnaribo_different_direction"),
                           rnapro = c("rnapro_enhanced",
                                      "rnapro_attenuated") ) )
summary_table_rnaribo_rnapro_consistency[1, 1] <- sum(ii_riboGTrna & ii_proGTrna)
summary_table_rnaribo_rnapro_consistency[1, 2] <- sum(ii_rnaGTribo & ii_rnaGTpro)
summary_table_rnaribo_rnapro_consistency[2, 1] <- 
    sum(ii_riboGTrna) - sum(ii_riboGTrna & ii_proGTrna)
summary_table_rnaribo_rnapro_consistency[2, 2] <- 
    sum(ii_rnaGTribo) - sum(ii_rnaGTribo & ii_rnaGTpro)

kable(summary_table_rnaribo_rnapro_consistency)</code></pre>
</div>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">rnapro_enhanced</th>
<th align="right">rnapro_attenuated</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">rnaribo_same_direction</td>
<td align="right">5</td>
<td align="right">84</td>
</tr>
<tr class="even">
<td align="left">rnaribo_different_direction</td>
<td align="right">104</td>
<td align="right">285</td>
</tr>
</tbody>
</table>
</div>
<h2 id="significance-tests">Significance tests</h2>
<h3 id="test-1">Test 1</h3>
<p>Compare the proportion of genes with attenuated translation versus enhanced translation (relative to RNA level), of the genes with sig. difference in translation efficiency between human and chimp AND sig. differential expression at the RNA levels between human and chimp.</p>
<p>Null hypothesis is that the proportion of attenuated translation and the proportion of enhanced translation relative to RNA level are not different, or equivalently, 50% of the genes of sig. translation efficiency difference between human and chimp AND DE at the RNA levels are either attenuated or enhanced at the translation level.</p>
<p>We propose to use a one-sample test of proportions to quantify the statistical significant of the proportion of genes attenuated (or enhanced) at translation.</p>
<p>Go here for a brief introduction to the one-proportion z-test: https://en.wikipedia.org/wiki/Statistical_hypothesis_testing.</p>
<p>Note that we use one-proportion z-test instead of an exact binomial test because of the large sample size. In fact the results are the</p>
<div class="row">
<div class="panel panel-primary">
<div class="source R panel-heading toggle">
<h5 class="panel-title">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</h5>
</div>
<pre style=""><code class="source r">no_samples <- sum(ii_sig_rnaribo)
prop.test(sum(ii_rnaGTribo), no_samples)</code></pre>
</div>
<div class="panel panel-success">
<div class="output R panel-heading toggle">
<h5 class="panel-title">
<span class="glyphicon glyphicon-chevron-down"></span> R output
</h5>
</div>
<pre style=""><code class="output r">## 
##  1-sample proportions test with continuity correction
## 
## data:  sum(ii_rnaGTribo) out of no_samples, null probability 0.5
## X-squared = 140.34, df = 1, p-value < 2.2e-16
## alternative hypothesis: true p is not equal to 0.5
## 95 percent confidence interval:
##  0.7311758 0.8083009
## sample estimates:
##         p 
## 0.7719665
</code></pre>
</div>
</div>
<h3 id="test-2">Test 2</h3>
<p>Alternative hypothesis: the effect of translation efficiency is more likely to be reflected at the protein level when translation is attenuated relative to transcript levels than reinforced.</p>
<p>Null hypothesis: Let p1 be the proportion of genes enhanced at protein level in the set of genes enhanced at the translation level, and p2 be the proportion of genes attenuated at translation level in the set of genes attenuated of translation level. Then, the null is p1 = p2, or equivalently, direction of protein divergence is consistent with direction of translation divergence.</p>
<p>Go here for a brief introduction to the two-proportion z-test: https://en.wikipedia.org/wiki/Statistical_hypothesis_testing.</p>
<div class="row">
<div class="panel panel-primary">
<div class="source R panel-heading toggle">
<h5 class="panel-title">
<span class="glyphicon glyphicon-chevron-down"></span> R source
</h5>
</div>
<pre style=""><code class="source r">## Use a Humanzee function
p_value <- PropTest(84/369, 5/109, N1 = 369, N2 = 109)

results <- c("p-value", p_value )

kable( results )</code></pre>
</div>
<table>
<tbody>
<tr class="odd">
<td align="left">p-value</td>
<td align="right">4.283458</td>
<td align="right">1.84e-05</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="navbar navbar-fixed-bottom navbar-inverse">
<div class="container">
<div class="navbar-header">
<button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-responsive-collapse">
<span class="icon-bar"></span> <span class="icon-bar"></span> <span class="icon-bar"></span>
</button>
</div>
<div id="bottom-navbar" class="navbar-collapse collapse navbar-responsive-collapse">
<ul class="nav navbar-nav navbar-right">
<li class="nav">
<p class="navbar-text">
Toggle
</p>
</li>
<li class="dropup">
<a href="#" class="dropdown-toggle" data-toggle="dropdown">Code <b class="caret"></b></a>
<ul class="dropdown-menu">
<li class="dropdown-header">
Languages
</li>
<li class="active">
<a href="#" class="toggle-global source R" type="source.R">R</a>
</li>
<li>
<a href="#" type="all-source" class="toggle-global">All</a>
</li>
</ul>
</li>
<li class="dropup">
<a href="#" class="dropdown-toggle" data-toggle="dropdown">Output <b class="caret"></b></a>
<ul class="dropdown-menu">
<li class="dropdown-header">
Type
</li>
<li class="active">
<a href="#" class="toggle-global output" type="output">output</a>
</li>
<li>
<a href="#" type="all-output" class="toggle-global">All</a>
</li>
</ul>
</li>
<li class="active">
<a href="#" type="figure" class="toggle-global">Figures</a>
</li>
</ul>
</div>
</div>
</div>
</div>
<div id="push">

</div>
<div id="footer">
<div class="container">
<p class="text-muted" id="credit">
Styled with <a href="https://github.com/jimhester/knitrBootstrap">knitrBootstrap</a>
</p>
</div>
</div>
<link rel="stylesheet" id="theme" href="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css" media="screen"></link><link rel="stylesheet" id="highlight" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/7.3/styles/default.min.css" media="screen"></link>
</div>
</body>
</html>
